src/client/dfs/duns.c

```diff
@@ -314,6 +331,11 @@ duns_resolve_path(const char *path, struct duns_attr_t *attr)
 			return EINVAL;
 		}
 
+		if (pool_only) {
+			D_FREE(dir);
+			return 0;
+		}
+
```

Comment:
Consider this input, where a pool is given, no container, but there is a relative path:
```
daos://67169544-c445-4069-bfd0-14f5e87c7209/path
```
Then this is the result
```
duns_resolve_path returnd 0
dattr.da_puuid    = 67169544-c445-4069-bfd0-14f5e87c7209
dattr.da_cuuid    = 00000000-0000-0000-0000-000000000000
dattr.da_rel_path = (null)
```

A path should maybe be invalid when `pool_only`?

Reply:
that's up to the user.. don't see why it matters here whether rel_path is parsed or not

Reply:
I guess I would expect one of these two scenarios:
1. `duns_resolve_path` returns an error, because a path relative to pool doesn't make sense.
2. `duns_resolve_path` copies the relative path into `da_rel_path`.

Essentially, right now, the `path` that was passed in is just "gone". It's neither used nor seen as invalid.

An example is from the perspective of middleware (such as the datamover utilities) that take a path from a user at the command line and pass this to `duns_resolve_path`. It's possible that a user just accidentally forgot type the container uuid in the path. So unless those middleware do some extra custom parsing, it's not possible to let the user know that their input was invalid. Seemingly, if a `path` was supplied, the user must have meant for "something" to be done with it.

Anyway, I'm not stuck on this, so I'll let you decide whether it's a valid point.

Reply:
@daltonbohning @mchaarawi I guess this makes it hard for us to tell whether or not duns_resolve_path was valid or not in the data mover tools. We check if duns_resolve_path == 0, then we assume it is a daos path. In this case one of our copies would continue without knowing it was actually an invalid path. If there is a path specified we would need to mount dfs, which means we need a valid container, and in that case it should fail. I think it just depends on whether we want to report that path as "invalid" or not from duns_resolve_path. 

This will also simplify the cont clone since now we can use this function when no cont UUID is specified.

Reply:
@dsikich In our use case, I think what would happen is that this:
```
dattr.da_puuid    = 67169544-c445-4069-bfd0-14f5e87c7209
dattr.da_cuuid    = 00000000-0000-0000-0000-000000000000
dattr.da_rel_path = (null)
```
Would see the input as just a pool, and we would end up creating a NEW container, and the path would just be the root of that new container.
So I don't think it would _**fail**_ - we just might end up copying to the root of a new container.

Reply:
I.e. if a user passes this path:
```
daos://67169544-c445-4069-bfd0-14f5e87c7209/path
```
Then the tools would create a NEW container in `pool=67169544-c445-4069-bfd0-14f5e87c7209`, and copy to the ROOT of that new container.

Reply:
so you would rather have this fail in this case? i can make that change if it helps the data mover.
but the thing is i can't have this fail:
daos://67169544-c445-4069-bfd0-14f5e87c7209/
since if we want to use the direct format with HDF5 it parses the path with dirname and passes that to get the pool.

Reply:
It WOULD help if this fails, because we can let the user know their input is invalid:
```
daos://67169544-c445-4069-bfd0-14f5e87c7209/path
```
It would help if this does NOT fail, because the user might want US to create a new container:
```
daos://67169544-c445-4069-bfd0-14f5e87c7209/
```

Reply:
ok then we are on the same page. i will make the change

Reply:
Thank you!

Reply:
> @dsikich In our use case, I think what would happen is that this:
> 
> ```
> dattr.da_puuid    = 67169544-c445-4069-bfd0-14f5e87c7209
> dattr.da_cuuid    = 00000000-0000-0000-0000-000000000000
> dattr.da_rel_path = (null)
> ```
> 
> Would see the input as just a pool, and we would end up creating a NEW container, and the path would just be the root of that new container.
> So I don't think it would _**fail**_ - we just might end up copying to the root of a new container.

@daltonbohning Yes that's true, it just would fail to write to some container they may have meant to specify if they are using an already existing path in the container. At least it would not overwrite their data in an already existing container or something.